client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -507,9 +512,18 @@ impl GenericProto {
 	///
 	/// Can be called multiple times with the same `PeerId`s.
 	pub fn add_discovered_nodes(&mut self, peer_ids: impl Iterator<Item = PeerId>) {
-		self.peerset.discovered(peer_ids.map(|peer_id| {
+		let local_peer_id = &self.local_peer_id;
+		self.peerset.discovered(peer_ids.filter_map(|peer_id| {
+			if peer_id == *local_peer_id {
+				error!(
```

Comment:
Should this really be printed as `error` if it is not that important?

Reply:
The problem by itself doesn't have any negative consequence, but it is a hint that the discovery system isn't working properly.